.github/workflows/odh-build-and-publish-operator-image.yaml

```diff
@@ -0,0 +1,106 @@
+# This is a copy of the publish-core-images.yaml and has been customized to
+# use the quay login credentials.
+# The unused parts of the original have been commented out on purpose.
+name: ODH -- Build and Publish Training Operator Images
+
+on:
+  - push
+  - pull_request
```

Comment:
should this be used on PR creation too?

Reply:
As far as I know yes. The image will be build and it will not be pushed. The quay login step has a condition that only allows for images to be build/pushed if the workflow is executed on default branch (dev) or a release branch or push of tag.

Reply:
correct.
what i was thinking: 
upon condition "pull_request" GH action only servers case for "go build" on 2 different arch.
the steps of "login quey" "get image meta" even "build image" are not really useful. 
so why not changed the workflow:
do the common parts at first, then stop it if it is only "non-dev/release/tag" condition
the rest condition can go with  steps like login quay + meta +  buildah etc.

Reply:
I have moved the steps a bit in: [10e7ab9](https://github.com/opendatahub-io/training-operator/pull/4/commits/10e7ab9d1a25e4a88ce167c09935f833a06d7208). The structure should be more logical now (I hope).